Userland/Utilities/su.cpp

```diff
@@ -21,37 +21,50 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     if (!TRY(Core::System::isatty(STDIN_FILENO)))
         return Error::from_string_literal("Standard input is not a terminal");
 
-    char const* user = nullptr;
+    StringView first_positional;
+    StringView second_positional;
+    bool simulate_login = false;
 
     Core::ArgsParser args_parser;
-    args_parser.add_positional_argument(user, "User to switch to (defaults to user with UID 0)", "user", Core::ArgsParser::Required::No);
+    args_parser.add_positional_argument(first_positional, "See --login", "-", Core::ArgsParser::Required::No);
+    args_parser.add_positional_argument(second_positional, "User to switch to (defaults to user with UID 0)", "user", Core::ArgsParser::Required::No);
+    args_parser.add_option(simulate_login, "Simulate login", "login", 'l');
     args_parser.parse(arguments);
 
+    StringView user = first_positional;
+
+    if (first_positional == '-') {
+        simulate_login = true;
+        user = second_positional;
+    }
+
     if (geteuid() != 0)
         return Error::from_string_literal("Not running as root :(");
 
-    auto account = TRY(user ? Core::Account::from_name(user) : Core::Account::from_uid(0));
+    auto account = TRY(user.is_null() ? Core::Account::from_uid(0) : Core::Account::from_name(user));
```

Comment:
`String::is_null()` and `StringView::is_null()` are things we're trying to replace with `Optional<>` or just "empty" strings. `ArgsParser` seems to not yet support giving it an `Optional<StringView>`, so I'd suggest initializing `first_positional` and `second_positional` to `""sv` and then using `is_empty()`. (Unless you feel like adding the support to ArgsParser. :yakbait:)

Reply:
I will change it to use `is_empty()`.
I think that this PR already has enough side commits, but I will have a look at `ArgsParser` :yakplus: